src/tsickle.ts

```diff
@@ -2017,6 +2025,51 @@ export function emitWithTsickle(
   };
 }
 
+function addClutzAliases(
+    dtsFileContent: string, sourceFiles: ts.SourceFile[], typeChecker: ts.TypeChecker,
+    host: TsickleHost): string {
+  let reexports = '';
+  for (const sf of sourceFiles) {
+    const moduleSymbol = typeChecker.getSymbolAtLocation(sf);
+    // const moduleExports = moduleSymbol && moduleSymbol.exports;
+    const moduleExports = moduleSymbol && typeChecker.getExportsOfModule(moduleSymbol);
+
+    if (!moduleExports) {
+      return dtsFileContent;
+    }
+
+    // moduleExports is a ts.Map<ts.Symbol> which is an es6 Map, but has a
+    // different type for no reason
+    for (const symbol of moduleExports) {
+      // Want to alias the symbol to match what clutz would produce
+      // see examples at
+      // https://github.com/angular/clutz/tree/master/src/test/java/com/google/javascript/clutz
+      const clutzSymbolName = 'module$contents$' +
+          host.pathToModuleName('', sf.fileName).replace(/\./g, '$') + '_' + symbol.name;
+
+      const symbolType = typeChecker.getTypeAtLocation(symbol.valueDeclaration!);
+      console.log(symbolType);
+      if (symbolType.flags & ts.TypeFlags.Object &&
+          (symbolType as ts.ObjectType).objectFlags & ts.ObjectFlags.Class) {
```

Comment:
iirc there's a way to ask TypeScript whether a symbol is a value or only a type. Check the code base – we do make that decision somewhere else already, I think related to dereferencing `*` exports.

Reply:
Yep, made this code much more elegant.